src/main/java/utils/MobileContextUtils.java

```diff
@@ -0,0 +1,85 @@
+package utils;
+
+import com.qaprosoft.carina.core.foundation.webdriver.DriverHelper;
+import io.appium.java_client.AppiumDriver;
+import org.apache.log4j.Logger;
+import org.openqa.selenium.NotFoundException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.events.EventFiringWebDriver;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Proxy;
+import java.util.Set;
+
+public class MobileContextUtils {
+
+    private static final Logger LOGGER = Logger.getLogger(DriverHelper.class);
+
+    private static AppiumDriver getDriverSafe(WebDriver driver) {
```

Comment:
There is will be problem with multithreading

Reply:
I believe it would be much more better.
public WebDriver getDriverSafe() {
        WebDriver driver = getDriver();
        if (driver instanceof EventFiringWebDriver) {
            driver = ((EventFiringWebDriver) drv).getWrappedDriver();
        }
        return driver;
    }